eve/methods/delete.py

```diff
@@ -193,20 +193,34 @@ def delete(resource, **lookup):
 
     .. versionadded:: 0.0.2
     """
-    getattr(app, "on_delete_resource")(resource)
-    getattr(app, "on_delete_resource_%s" % resource)()
 
     resource_def = config.DOMAIN[resource]
+    getattr(app, "on_delete_resource")(resource)
+    getattr(app, "on_delete_resource_%s" % resource)()
+    default_request = ParsedRequest()
+    if resource_def['soft_delete']:
+        # get_document should always fetch soft deleted documents from the db
+        # callers must handle soft deleted documents
+        default_request.show_deleted = True
```

Comment:
this behavior should probably be explained in the [Soft Delete](http://python-eve.org/features.html#soft-delete) section of the docs. Explain that the new callback will include soft-deleted documents when soft delete is active.